packages/component-submission/client/pages/SubmissionWizard.test.js

```diff
@@ -1,109 +1,269 @@
-      history: jest.fn(),
-      initialValues: { fileStatus: 'READY' },
-    }
+    const renderWithPath = path =>
+      render(<SubmissionWizard {...makeProps(path)} />, {
+        wrapper: setup([path]),
+      })
 
-    const thing = shallow(<SubmissionWizard {...props} />)
+    // disable formik warnings
+    // see https://stackoverflow.com/questions/55181009/jest-react-testing-library-warning-update-was-not-wrapped-in-act
+    // eslint-disable-next-line no-console
+    const consoleError = console.error
+    // eslint-disable-next-line no-console
+    console.error = (...args) => {
+      if (
+        !args[0].startsWith(
+          'Warning: An update to Formik inside a test was not wrapped in act(...).',
+        )
+      ) {
+        consoleError(...args)
+      }
+    }
 
-    expect(thing.instance().state.isUploading).toBe(false)
-    expect(thing.instance().state.suspendSave).toBe(false)
+    opts = renderWithPath('/submit/id/author')
+    fireEvent.click(opts.getByTestId('next'))
+    await flushPromises()
+    expect(pushHistory).toHaveBeenCalledTimes(1)
+    expect(pushHistory.mock.calls[0]).toEqual(['/files'])
+    cleanup()
+    pushHistory.mockReset()
 
-    thing.instance().setIsUploading(true)
+    opts = renderWithPath('/submit/id/files')
+    fireEvent.click(opts.getByTestId('next'))
+    await flushPromises()
+    expect(pushHistory).toHaveBeenCalledTimes(1)
+    expect(pushHistory.mock.calls[0]).toEqual(['/details'])
+    cleanup()
+    pushHistory.mockReset()
 
-    expect(mockUpdate).toHaveBeenCalledTimes(0)
+    opts = renderWithPath('/submit/id/details')
+    fireEvent.click(opts.getByTestId('next'))
+    await flushPromises()
+    expect(pushHistory).toHaveBeenCalledTimes(1)
+    expect(pushHistory.mock.calls[0]).toEqual(['/editors'])
+    cleanup()
+    pushHistory.mockReset()
 
-    thing.instance().onNextClick({ submitted_should_be: false })
-    thing.instance().setIsUploading(false)
+    opts = renderWithPath('/submit/id/editors')
+    fireEvent.click(opts.getByTestId('next'))
+    await flushPromises()
+    expect(pushHistory).toHaveBeenCalledTimes(1)
+    expect(pushHistory.mock.calls[0]).toEqual(['/disclosure'])
+    cleanup()
+    pushHistory.mockReset()
 
-    expect(mockUpdate).toHaveBeenCalledTimes(1)
-    expect(mockUpdate.mock.calls).toEqual([[{ submitted_should_be: false }]])
-    expect(thing.instance().state.nextSaveValues).toBeNull()
+    // eslint-disable-next-line no-console
+    console.error = consoleError
   })
 
-  it('disables the autosave on submit and re-enables it on completion', async () => {
-    const mockUpdate = jest.fn()
-    const mockSubmit = jest.fn(() =>
-      new Promise(resolve => {
-        expect(thing.instance().state.suspendSave).toBe(true)
-        resolve()
-      }).catch(err => {
-        // eslint-disable-next-line no-undef
-        fail(err)
-      }),
+  it('should change the location on back except on first step', async () => {
+    let opts = null
```

Comment:
Again, why reuse the same variable? I'll be much more readable if it's a more descriptive name per test case